students/surowiec_marek/lesson_02_flow_control/lesson_02_flow_control.py

```diff
@@ -0,0 +1,125 @@
+# Bishop moves
+h1 = int(input())
+v1 = int(input())
+h2 = int(input())
+v2 = int(input())
+if (abs(h1-h2) == abs(v1-v2)):
+    print('YES')
+else:
+    print('NO')
+
+
+# Queen move
+h1 = int(input())
+v1 = int(input())
+h2 = int(input())
+v2 = int(input())
+if (abs(h1-h2) == abs(v1-v2) or h1 == h2 or v1 == v2):
+    print('YES')
+else:
+    print('NO')
+
+# Knight move
+h1 = int(input())
+v1 = int(input())
+h2 = int(input())
+v2 = int(input())
+if (abs(h1-h2) == 1 and abs(v1-v2) == 2 or
+    abs(h1-h2) == 2 and abs(v1-v2) == 1):
+    print('YES')
+else:
+    print('NO')
+
+
+# Chocolate bar
+n = int(input())
+m = int(input())
+k = int(input())
+if (k % n == 0 and k/n < m) or (k % m == 0 and k/m < n):
+    print('YES')
+else:
+    print('NO')
+
+# Leap year
+year = int(input())
+if (year % 400 == 0 or year % 4 == 0 and year % 100 != 0):
+    print('LEAP')
+else:
+    print('COMMON')
+
+
+# Factorial
+factorial = 1
+n = int(input())
+for i in range(1, n+1):
+    factorial *= i
+print(factorial)
+
+
+# The number of zeros
+zero_count = 0
+N = int(input())
+for i in range(N):
+    n = int(input())
+    if n == 0:
+        zero_count += 1
+print(zero_count)
+
+
+# Adding factorials
+factorial = 1
+sum = 0
```

Comment:
`sum` jest funkcją built-in (tak jak np. `max`). W pythonie w odróżnieniu od np. Javy funkcje są obiektami i możesz je dowolnie przypisywać do zmiennych. Oznacza to, że możesz sobie coś nadpisać: 
```
>>> max
<built-in function max>
>>> moj_max = max <- przypisałem funkcję max do moj_max
>>> moj_max
<built-in function max>
>>> max = 0 <- przypisałem 0 do max
>>> max
0
```
Jak masz wątpliwości czy nazwa jest wolna to możesz sprawdzić w konsoli pythona:
```
>>> max                                       
<built-in function max>                       
>>> moja_zmienna                              
Traceback (most recent call last):            
  File "<stdin>", line 1, in <module>         
NameError: name 'moja_zmienna' is not defined 
>>>                                           
```

Zalecam poprawkę, żeby przypadkowo nie przesłaniać built-inów